src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -35,6 +41,30 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
   const [quantity, setQuantity] = useState<number | undefined>(undefined);
   const [schematicDesignation, setSchematicDesignation] = useState<string | undefined>(undefined);
 
+  const [snsPredictedFailureRate, setSnsPredictedFailureRate] = useState<number | undefined>(undefined);
+  const [snsManuallyDefinedFailureRate, setSnsManuallyDefinedFailureRate] = useState<number | undefined>(undefined);
+  const [selectedRadioButton, setSelectedRadioButton] = useState<string>(RadioButtonType.Predicted);
+
+  const handleManuallyDefinedFailureRateChange = (event) => {
+    const inputValue = event.target.value;
+    const regex = /^[0-9]*\.?[0-9]*$/;
+    if (regex.test(inputValue)) {
```

Comment:
I'm not sure what will happen if the regex fails. Do we just silently ignore it?  If so, create a separate issue to resolve this. 

BTW, will we have manual failure rate probability selected while not having a value ? 
Maybe a workaround is to make probability equal to 1 and mark the selected value as red to indicate it.

Reply:
Will be resolved by https://github.com/kbss-cvut/fta-fmea-ui/issues/407